### PR TITLE
[9.x] Option to set UUID/ULID as primary key on migration and model files

### PR DESCRIPTION
**Pull request by morcen**
Since https://github.com/laravel/framework/pull/44146 we now have options to use UUID and ULID as the table's primary key. This PR aims to assist developers in setting up UUID or ULID during migration and model generation, following the instructions of the [official documentation](https://laravel.com/docs/9.x/eloquent#uuid-and-ulid-keys).

## How to use:
Use the option `--primary` (or shortcut `-P`) to define the type of primary key of the migration file or model being created.

Screenshot 2022-12-19 at 11 17 03 pm

For example:
- To generate a migration file that creates the table `books` with primary key type `UUID`:
    `php artisan make:migration create_books_table -Puuid`
    Screenshot 2022-12-19 at 11 33 44 pm
    Screenshot 2022-12-19 at 11 36 43 pm

- To generate a `Customer` model that also creates the migration file with primary key type `ULID`:
    `php artisan make:model Customer -mPulid`
Screenshot 2022-12-19 at 11 34 28 pm
Screenshot 2022-12-19 at 11 35 43 pm
Screenshot 2022-12-19 at 11 35 56 pm

### More examples
- Generate the model `Student` with `-a` (all) options, and set the primary key to UUID:
    `php artisan make:model Student --all --primary=uuid`
    Screenshot 2022-12-19 at 11 42 07 pm
    Screenshot 2022-12-19 at 11 42 30 pm
    Screenshot 2022-12-19 at 11 43 07 pm

Changes in this PR will not affect those who wish to use the default auto-incrementing integer as the primary key, as simply omitting the `primary` option (or leaving it empty) will result in the default PK.

### CODE REVIEW

File: src/Illuminate/Database/Migrations/MigrationCreator.php [hunk 1]
```diff
@@ -209,6 +214,23 @@ protected function getDatePrefix()
...
+    protected function getPrimaryKeyField($primary)
+    {
+        if ($primary === 'UUID') {
+            return 'uuid(\'id\')';
+        }
+        if ($primary === 'ULID') {
+            return 'ulid(\'id\')';
+        }
+
+        return 'id()';
```
Comment (crynobone) [inline src/Illuminate/Database/Migrations/MigrationCreator.php:231]: why not just utilise `Illuminate\Database\Schema\Blueprint::$defaultMorphKeyType`

Comment (morcen) [reply src/Illuminate/Database/Migrations/MigrationCreator.php:228]: thanks @crynobone , I will look into utilising that instead

Comment (morcen) [reply src/Illuminate/Database/Migrations/MigrationCreator.php:226]: @crynobone, how do you suggest the `Blueprint::$defaultMorphKeyType` be used in this section?

File: src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php [hunk 2]
```diff
@@ -18,7 +19,8 @@ class MigrateMakeCommand extends BaseCommand
         {--table= : The table to migrate}
         {--path= : The location where the migration file should be created}
         {--realpath : Indicate any provided migration file paths are pre-resolved absolute paths}
-        {--fullpath : Output the full path of the migration}';
+        {--fullpath : Output the full path of the migration}
+        {--P|primary=int : Set default primary key type for the table <info>[int, uuid, ulid]</info>}';
```
Comment (morcen) [inline src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php:23]: `--key` might be shorter but I don't find it as descriptive as `--primary`. Thoughts?

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**